pandas/tseries/index.py

```diff
@@ -827,16 +828,21 @@ def _format_native_types(self, na_rep=u('NaT'),
     def to_datetime(self, dayfirst=False):
         return self.copy()
 
-    def astype(self, dtype):
+    def astype(self, dtype, copy=True):
         dtype = np.dtype(dtype)
```

Comment:
`CategoricalIndex` is making some troubles because it doesn't have `.astype()` and `Categorical.astype` doesn't have a parameter `copy`. So, which solution is better:

1. Add a parameter `copy` to `Categorical.astype()`, which now looks like that:
    ```python
    def astype(self, dtype):
        """ coerce this type to another dtype """
        if is_categorical_dtype(dtype):
            return self
        return np.array(self, dtype=dtype)
    ```

   (If so, what should be default for `copy`?)

2. Create `CategoricalIndex.astype()` which would probably double the code from `Index.astype()` ignoring `copy`.

3. Catch an exception in `Index.astype()`:
    ```python
    def astype(self, dtype, copy=True):
        try:
            result = Index(self.values.astype(dtype, copy=copy),
                           name=self.name, dtype=dtype)
        except:
            result = Index(self.values.astype(dtype), name=self.name,
                     dtype=dtype)
        return result
    ```